### ISSUE TITLE
html-formatter: Cannot generate HTML report from NDJSON file

### ISSUE THREAD

**Issue by x80486** [cucumber/cucumber#1241]
## Summary

I'm using `@cucumber/cucumber` version `7.0.0-rc.0` with Node.js `14.15.0`. I installed `cucumber-html-formatter` globally (as instructed in the `README`), and after running the tests that generates the NDJSON file, the report fails.

## Expected Behavior

The report should be generated.

## Current Behavior

The report fails with:

```shell
[x80486@uplink:~/ccbr]$ cat reports/cucumber-messages.ndjson | cucumber-html-formatter --format ndjson > reports/index.html 

/home/x80486/.asdf/installs/nodejs/14.15.0/.npm/lib/node_modules/cucumber-html-formatter/node_modules/cucumber-query/src/CucumberQuery.ts:200
      if (message.testCaseFinished.testResult.status === undefined) {
                                              ^
TypeError: Cannot read property 'status' of null
    at CucumberQuery.update (/home/x80486/.asdf/installs/nodejs/14.15.0/.npm/lib/node_modules/cucumber-html-formatter/node_modules/cucumber-query/src/CucumberQuery.ts:200:47)
    at /home/x80486/.asdf/installs/nodejs/14.15.0/.npm/lib/node_modules/cucumber-html-formatter/src/cli-main.tsx:41:21
    at Array.reduce (<anonymous>)
    at /home/x80486/.asdf/installs/nodejs/14.15.0/.npm/lib/node_modules/cucumber-html-formatter/src/cli-main.tsx:40:44
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:63:3)
```

## Possible Solution

No idea :sweat_smile:

## Steps to Reproduce (for bugs)

1. Set up a Cucumber.js (TypeScript `4.0.5`) project
1. Install `cucumber-html-formatter` globally via `NPM`
1. Add `@cucumber/cucumber` version `7.0.0-rc.0`
1. Put some dummy test(s) in order to generate a `cucumber-messages.ndjson` file
1. Execute `cat /path/to/cucumber-messages.ndjson | cucumber-html-formatter --format ndjson > index.html`

## Context & Motivation

Trying to see if the generated report is good enough to replace `cucumber-html-reporter`.

## Your Environment

* Version used: `@cucumber/cucumber` version `7.0.0-rc.0`
* Operating System and version: `Linux uplink 5.4.74-1-lts #1 SMP Sun, 01 Nov 2020 12:58:27 +0000 x86_64 GNU/Linux`
* Link to your project: `N/A`

**Comment by aslakhellesoy** [comment]
>  I installed cucumber-html-formatter globally (as instructed in the README)

What README is instructing you to do that?

The correct way to generate a HTML report with cucumber-js is to use `--format html`.

**Comment by x80486** [comment]
I was referring to the `README` at this location: https://github.com/cucumber/cucumber/tree/master/html-formatter/

Moreover, `--format html` doesn't do anything for me, and `--format html:reports/cucumber-report.html`

```shell
[x80486@uplink:~/ccbr]$ npm run test 

> ccbr@0.2.0 pretest /home/x80486/ccbr
> mkdirp reports/


> ccbr@0.2.0 test /home/x80486/ccbr
> cucumber-js --profile default

Error: Cannot find module '/home/x80486/ccbr/html'
Require stack:
- /home/x80486/ccbr/node_modules/@cucumber/cucumber/lib/formatter/builder.js
- /home/x80486/ccbr/node_modules/@cucumber/cucumber/lib/cli/index.js
- /home/x80486/ccbr/node_modules/@cucumber/cucumber/lib/cli/run.js
- /home/x80486/ccbr/node_modules/@cucumber/cucumber/bin/cucumber-js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._resolveFilename (/home/x80486/ccbr/node_modules/tsconfig-paths/lib/register.js:75:40)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.loadCustomFormatter (/home/x80486/ccbr/node_modules/@cucumber/cucumber/src/formatter/builder.ts:107:29)
    at Object.getConstructorByType (/home/x80486/ccbr/node_modules/@cucumber/cucumber/src/formatter/builder.ts:81:33)
    at Object.build (/home/x80486/ccbr/node_modules/@cucumber/cucumber/src/formatter/builder.ts:42:51)
    at /home/x80486/ccbr/node_modules/@cucumber/cucumber/src/cli/index.ts:129:31
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ccbr@0.2.0 test: `cucumber-js --profile default`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ccbr@0.2.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/x80486/.npm/_logs/2020-11-09T11_45_25_442Z-debug.log
```

**Comment by hdorgeval** [comment]
I have the same problem as above. I am running a typescript project, the cucumber command-line has : `--format html:reports/report.html`, but the html file is empty and I have the same type of error:

```sh
> cucumber-ts-starter@4.0.0 cucumber /Users/HDO/VSCodeProjects/cucumber-ts-starter
> node ./node_modules/@cucumber/cucumber/bin/cucumber-js features/**/*.feature --require env/set-environment-variables.ts --require world/custom-world.ts --require step-definitions/**/*.ts --require hooks/**/*.ts  --require-module ts-node/register --format-options "{\"snippetInterface\": \"async-await\"}" --format json:reports/cucumber-report.json --format html:reports/report.html --format summary --format progress-bar --publish-quiet

The built-in JSON formatter is deprecated and will be removed in the next major release. Where you need a structured data representation of your test run, it's best to use the `message` formatter. For legacy tools that depend on the deprecated JSON format, a standalone formatter is available (see https://github.com/cucumber/cucumber/tree/master/json-formatter).
Error: Cannot find module '/Users/HDO/VSCodeProjects/cucumber-ts-starter/html'
Require stack:
- /Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/lib/formatter/builder.js
- /Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/lib/cli/index.js
- /Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/lib/cli/run.js
- /Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/bin/cucumber-js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.loadCustomFormatter (/Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/src/formatter/builder.ts:107:29)
    at Object.getConstructorByType (/Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/src/formatter/builder.ts:81:33)
    at Object.build (/Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/src/formatter/builder.ts:42:51)
    at /Users/HDO/VSCodeProjects/cucumber-ts-starter/node_modules/@cucumber/cucumber/src/cli/index.ts:129:31
```

**Comment by aslakhellesoy** [comment]
I believe this should be fixed in 7.0.0 - reopening this so we can confirm when it’s released.

**Comment by davidjgoss** [comment]
7.0.0 is released and `--format html` now works - closing but please feed back if any specific issues with the formatter.
